app/services/borrow_direct_reshare_client.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require 'http'
+
+class BorrowDirectReshareClient
+  DEFAULT_HEADERS = {
+    accept: 'application/json, text/plain',
+    content_type: 'application/json'
+  }.freeze
+
+  attr_reader :base_url
+
+  def initialize(url: Settings.borrow_direct_reshare.url,
+                 username: Settings.borrow_direct_reshare.username,
+                 password: Settings.borrow_direct_reshare.password,
+                 tenant: Settings.borrow_direct_reshare.tenant)
+    uri = URI.parse(url)
+
+    @base_url = url
+    @username = username
+    @password = password
+
+    if uri.user
+      @username ||= uri.user
```

Comment:
Is there a reason we need to handle both the case where the user/pass are encoded in the URI and one where they are passed in?